common/config/environment.go

```diff
@@ -72,21 +71,65 @@ type CommonEnvironment struct {
 	username string
 }
 
+type Env interface {
+	provider
+
+	Ctx() *pulumi.Context
+	CommonNamer() namer.Namer
+
+	InfraShouldDeployFakeintakeWithLB() bool
+	InfraEnvironmentNames() []string
+	InfraOSDescriptor() string
+	InfraOSImageID() string
+	KubernetesVersion() string
+	DefaultResourceTags() map[string]string
+	ExtraResourcesTags() map[string]string
+	ResourcesTags() pulumi.StringMap
+
+	AgentDeploy() bool
+	AgentVersion() string
+	PipelineID() string
+	CommitSHA() string
+	ClusterAgentVersion() string
+	AgentFullImagePath() string
+	ClusterAgentFullImagePath() string
+	ImagePullRegistry() string
+	ImagePullUsername() string
+	ImagePullPassword() pulumi.StringOutput
+	AgentAPIKey() pulumi.StringOutput
+	AgentAPPKey() pulumi.StringOutput
+	AgentUseFakeintake() bool
+	TestingWorkloadDeploy() bool
+	DogstatsdDeploy() bool
+	DogstatsdFullImagePath() string
+	UpdaterDeploy() bool
+
+	GetBoolWithDefault(config *sdkconfig.Config, paramName string, defaultValue bool) bool
+	GetStringListWithDefault(config *sdkconfig.Config, paramName string, defaultValue []string) []string
+	GetStringWithDefault(config *sdkconfig.Config, paramName string, defaultValue string) string
+	GetObjectWithDefault(config *sdkconfig.Config, paramName string, outputValue, defaultValue interface{}) interface{}
+	GetIntWithDefault(config *sdkconfig.Config, paramName string, defaultValue int) int
```

Comment:
Actually all these methods are defined in `CommonEnvironment` that is embedded in `Environment` so we do not need to define these several times.
To me it is easier to use these functions directly as method of the environment. Did you have anything in mind that would make it better if we change it to static helpers?

Reply:
Thinking about them, I ask how much they depend on the environment state.

Looking at one of the implementation, those functions use the environment only to log, and the logger is in the environment context.

https://github.com/DataDog/test-infra-definitions/blob/e5e1a13ee2c4873f97c8b3e1cef7e85a0d405fc7/common/config/environment.go#L245-L256

We could pass the context, and skip the strong dependency with the environment, so that we could get any value wherever from the configuration

Reply:
Actually they do not depend on the environment. That's why they are method defined on `CommonEnvironment`, ([here](https://github.com/DataDog/test-infra-definitions/blob/main/common/config/environment.go#L60)) and every environment should embed `CommonEnvironment

Reply:
Sure! Looks like they need an interface that implements a `Log`, ok for having it later